### PR TITLE
Updating and refactoring GeoMechanicsApplication

### PR DESCRIPTION
**Pull request by Vahid-Galavi**
**📝 Description**
As part of improvement of GeoMechanicsApplication, several improvements have been made.

**🆕 Changelog**
Please summarize the changes in one list to generate the changelog:
E.g.
- Updated Lagrangian
- ImplicitSolvingStrategy instead of SolvingStrategy
- Tests are updated
- Bug in writing outputs

### CODE REVIEW

File: applications/GeoMechanicsApplication/custom_constitutive/small_strain_udsm_3D_law.cpp [hunk 1]
```diff
@@ -262,6 +240,22 @@ void SmallStrainUDSM3DLaw::InitializeMaterial(const Properties &rMaterialPropert
    // loading the model
    mIsUDSMLoaded = loadUDSM(rMaterialProperties);
 
+   if (!mIsUDSMLoaded) {
+      KRATOS_ERROR << "cannot load the specified UDSM " << rMaterialProperties[UDSM_NAME] << std::endl;
+   }
```
Comment (philbucher) [inline applications/GeoMechanicsApplication/custom_constitutive/small_strain_udsm_3D_law.cpp:245]:
```suggestion
   KRATOS_ERROR_IF_NOT(mIsUDSMLoaded) << "cannot load the specified UDSM " << rMaterialProperties[UDSM_NAME] << std::endl;
```

File: applications/GeoMechanicsApplication/custom_strategies/schemes/newmark_quasistatic_U_Pw_scheme.hpp [hunk 2]
```diff
@@ -244,8 +288,7 @@ class NewmarkQuasistaticUPwScheme : public Scheme<TSparseSpace,TDenseSpace>
     {
         KRATOS_TRY
 
-        if (rModelPart.GetProcessInfo()[NODAL_SMOOTHING] == true)
-        {
+        if (rModelPart.GetProcessInfo()[NODAL_SMOOTHING] == true) {
```
Comment (philbucher) [inline applications/GeoMechanicsApplication/custom_strategies/schemes/newmark_quasistatic_U_Pw_scheme.hpp:291]:
```suggestion
        if (rModelPart.GetProcessInfo()[NODAL_SMOOTHING]) {
```

File: applications/GeoMechanicsApplication/custom_elements/transient_Pw_element.cpp [hunk 3]
```diff
@@ -430,12 +418,11 @@ void TransientPwElement<TDim,TNumNodes>::
         if ( rOutput.size() != mRetentionLawVector.size() )
             rOutput.resize(mRetentionLawVector.size());
 
-        for ( unsigned int i = 0;  i < mRetentionLawVector.size(); i++ )
-        {
-            rOutput[i] = mRetentionLawVector[i]->GetValue( rVariable, rOutput[i] );
+        for ( unsigned int i = 0;  i < mRetentionLawVector.size(); ++i ) {
```
Comment (philbucher) [inline applications/GeoMechanicsApplication/custom_elements/transient_Pw_element.cpp:421]: you could use `std::fill` here (https://en.cppreference.com/w/cpp/algorithm/fill)

**Review by philbucher — APPROVED** [review]

**Comment by Vahid-Galavi** [comment]
@philbucher Thanks for your review and time. I will consider your suggestion in the next round of refactoring.

**Pull request merged**